### PR TITLE
Windows support

### PR DESCRIPTION
**Pull request by stimpy77**
## What this Pull Request (PR) does
Adds the missing Windows support without WSL. Applying these changes and then following the rest of the usual instructions, tested from my Powershell (not WSL):

```
❯ "2+2=?"|fabric
2 + 2 = 4
```

.. 

```
❯ yt https://www.youtube.com/watch?v=lE4opKy9fLY --transcript | fabric -sp summarize
ONE SENTENCE SUMMARY:
This paper introduces a revolutionary AIOS agent ecosystem, where large language models (LLMs) serve as the operating system, redefining user-developer interaction and software development through natural language programming.

MAIN POINTS:
1. Envisions LLMs as the core of an Artificial Intelligent Operating System (AIOS), a significant shift from traditional OS.
2. Proposes LLM-based AI agents (AAPS) to perform diverse tasks, .......
```

### CODE REVIEW

**Comment by danielmiessler** [comment]
Moving to Go soon; let's hold off on this one. But Thank You for submitting. Let's revisit soon!

**Comment by tonymet** [comment]
I'd like to nominate this to be merged.  Python still has a large audience, and rewrites always take longer than expected.    I'm happy to add test cases and help prepare this for merging.  cc: @danielmiessler

File: installer/client/cli/fabric.py [hunk 1]
```diff
@@ -75,11 +75,11 @@ def main():
                         help="Use Context file (context.md) to add context to your pattern", action="store_true")
 
     args = parser.parse_args()
-    home_holder = os.path.expanduser("~")
-    config = os.path.join(home_holder, ".config", "fabric")
-    config_patterns_directory = os.path.join(config, "patterns")
-    config_context = os.path.join(config, "context.md")
-    env_file = os.path.join(config, ".env")
+    home_holder = os.path.expanduser("~").replace("\\", "/")
+    config = os.path.join(home_holder, ".config", "fabric").replace("\\", "/")
```
Comment (tonymet) [inline installer/client/cli/fabric.py:79]: is this backward compatible with linux ?  os.path.join should handle OS-specific dir names

**Comment by stimpy77** [comment]
> @stimpy77 can you speak to the dir name replacements? e.g. `os.path.join(home_holder, ".config", "fabric").replace("\\", "/")`
> 
> I'd like to test on windows and see if we can help get this merged with fewer changes

Windows can handle forward slashes, Linux can't handle backslashes. I ran the changes on my Windows 11 machine before providing the PR. Everything worked great. As far as whether os.path.join(config, "context.md") should have already handled the issues, .. apparently not. But all changes were generated using a current GPT model (gpt-4o with cursor.sh), I just told it "make it Windows compatible".

**Comment by tonymet** [comment]
> > @stimpy77 can you speak to the dir name replacements? e.g. `os.path.join(home_holder, ".config", "fabric").replace("\\", "/")`
> > I'd like to test on windows and see if we can help get this merged with fewer changes
> 
> Windows can handle forward slashes, Linux can't handle backslashes. 

I'll pull your branch and see if I can help test this out.  `os.path.join` should be platform-independent and transform the slash accordingly

**Comment by tonymet** [comment]
@stimpy77 I was able to get fabric install on windows to work from the `main` branch . 


```
fabric --changeDefaultModel models/gemini-1.5-pro
❯ fabric --text "how many days in a year"

There are **365 days** in a regular year and **366 days** in a leap year.

>  yt --transcript 'https://www.youtube.com/watch?v=DhBstDpVIw4' | fabric  --pattern extract_wisdom
C:\Users\tonymet\pipx\venvs\fabric\Lib\site-packages\pydub\utils.py:170: RuntimeWarning: Couldn't find ffmpeg or avconv - defaulting to ffmpeg, but may not work
  warn("Couldn't find ffmpeg or avconv - defaulting to ffmpeg, but may not work", RuntimeWarning)
C:\Users\tonymet\pipx\venvs\fabric\Lib\site-packages\pydub\utils.py:170: RuntimeWarning: Couldn't find ffmpeg or avconv - defaulting to ffmpeg, but may not work
  warn("Couldn't find ffmpeg or avconv - defaulting to ffmpeg, but may not work", RuntimeWarning)
# OUTPUT
```

**Comment by tonymet** [comment]
I recommend merging the install notes.   With your install instructions and the following prerequisites I was able to install fabric.exe using pipx and run the workload as usual.

1. install Visual Studio 2022 with Python Native Workload https://learn.microsoft.com/en-us/visualstudio/python/working-with-c-cpp-python-in-visual-studio?view=vs-2022 or Visual Studio Build Tools Command line https://visualstudio.microsoft.com/downloads/#build-tools-for-visual-studio-2022

2. Make sure default model is set `fabric --changeDefaultModel models/gemini-1.5-pro`

File: installer/client/cli/fabric.py (hunk 1 above)
Comment (tonymet) [reply installer/client/cli/fabric.py:79]: after double checking, I was able to install from the same branch to both windows & linux using pipx without these modifications

**Comment by stimpy77** [comment]
I had issues with the main repos main branch on Windows without my functional changes. The fabric setup command failed. It puts binary links into the user .local profile. Maybe it was updated since I tried (date when I created the PR).

**Pull request closed without merging**